playwright/_impl/_element_handle.py

```diff
@@ -190,8 +191,18 @@ async def set_input_files(
         noWaitAfter: bool = None,
     ) -> None:
         params = locals_to_params(locals())
-        params["files"] = await normalize_file_payloads(files)
-        await self._channel.send("setInputFiles", params)
+        frame = await self.owner_frame()
+        if not frame:
+            raise Error("Cannot set input files to detached element")
+        converted = await convert_input_files(files, frame.page.context)
+        if converted.get("files") is not None:
+            params["files"] = converted.get("files")
+            await self._channel.send("setInputFiles", params)
+        else:
+            await self._channel.send(
+                "setInputFilePaths",
+                locals_to_params({**params, **converted, "files": None}),
```

Comment:
lets try to avoid locals_to_params here and instead either pass all the values or do it like above.